x/chainlink/client/rest/rest.go

```diff
@@ -18,10 +19,18 @@ func RegisterRoutes(clientCtx client.Context, r *mux.Router) {
 	registerTxHandlers(clientCtx, r)
 }
 
-func registerQueryRoutes(clientCtx client.Context, r *mux.Router) {
-	r.HandleFunc("chainlink/feed/data/{feedId}", listFeedDataHandler(clientCtx)).Methods(MethodGet)
+func registerQueryRoutes(clientCtx client.Context, rtr *mux.Router) {
+	r := clientrest.WithHTTPDeprecationHeaders(rtr)
+
+	r.HandleFunc("/chainlink/feed/data/{feedId}", listFeedDataByFeedIdHandler(clientCtx)).Methods(MethodGet) // query feed data by feedId
 }
 
-func registerTxHandlers(clientCtx client.Context, r *mux.Router) {
-	r.Handle("chainlink/feed/data", createFeedHandler(clientCtx)).Methods(MethodPOST)
+func registerTxHandlers(clientCtx client.Context, rtr *mux.Router) {
+	r := clientrest.WithHTTPDeprecationHeaders(rtr)
+
+	r.HandleFunc("/txs/encode", authrest.EncodeTxRequestHandlerFn(clientCtx)).Methods(MethodPOST)
+	r.HandleFunc("/txs/decode", authrest.DecodeTxRequestHandlerFn(clientCtx)).Methods(MethodPOST)
+	r.HandleFunc("/txs", authrest.BroadcastTxRequest(clientCtx)).Methods(MethodPOST)
+
+	r.Handle("/chainlink/feed/data", submitFeedDataHandler(clientCtx)).Methods(MethodPUT)
```

Comment:
why do we need PUT instead of POST here? jw

Reply:
PUT is used for submit, but it doesnt really `submit` data to store, instead it just returns a Std.Msg object, caller will need to use `broadcast` which is a POST to modify store